README.md

```diff
@@ -1,2 +1,281 @@
+      id:                                <computed>
+      accessor:                          <computed>
+      default_lease_ttl_seconds:         "2592000"
+      max_lease_ttl_seconds:             "2592000"
+      path:                              "database"
+      type:                              "database"
+
+
+Plan: 4 to add, 0 to change, 0 to destroy.
+```
+
+#### Terraform apply
+
+If the `terraform plan` output looks similar to the above one then run `terraform apply` to create the 
+resources.
+
+#### Trying it out
+
+After applying the changes, we should be able to use vault CLI to interact with our newly created database 
+secret backend:
+
+```bash
+# Ensure `VAULT_TOKEN` and `VAULT_ADDR` are set for this work!
+vault read database/config/postgres-secret-backend
+
+Key                                   Value
+---                                   -----
+allowed_roles                         [service-write dev-read]
+connection_details                    map[connection_url:postgres://postgres:*****@database:5432/postgres?sslmode=disable max_open_connections:2]
+plugin_name                           postgresql-database-plugin
+```
+
+##### Use write credentials
+
+First generate a new set of creds with vault:
+
+```bash
+vault read database/creds/service-write
+
+Key                Value
+---                -----
+lease_id           database/creds/service-write/lVpzrysA5akqSvjZVtCgx1i9
+lease_duration     240h
+lease_renewable    true
+password           A1a-9yW06ZdVk54I5KnX
+username           v-token-service--1luYzAYl7SdMxvdpibYv-1555972312
+```
+
+Then use them to login into your local Postgres database and make some changes:
+
+```bash
+psql -h localhost -U v-token-service--1luYzAYl7SdMxvdpibYv-1555972312 -W -d postgres
+
+CREATE TABLE test_table (
+             product_no integer,
+             name text,
+             price numeric
+         );
+INSERT INTO test_table VALUES (1, 'Cheese', 9.99);
+
+SELECT * FROM test_table;
+
+\dt;
+```
+
+You can also view your newly created user when running `\du;`
+
+##### Use read credentials
+
+Generate new creds and use them:
+```bash
+vault read database/creds/dev-read
+Key                Value
+---                -----
+lease_id           database/creds/dev-read/TLCjNu5vT73r0QCW6X86f26Z
+lease_duration     720h
+lease_renewable    true
+password           A1a-GiwY6h7CHbsMHLLL
+username           v-token-analytic-09puAVVxuQm6ELgrIMXT-1555973788
+```
+
+```bash
+psql -h localhost -U v-token-analytic-09puAVVxuQm6ELgrIMXT-1555973788 -W -d postgres
+
+# We can SELECT but not INSERT
+SELECT * FROM test_table;
+
+INSERT INTO test_table VALUES (2, 'Lettuce', 2.92);
+> ERROR:  permission denied for relation test_table
+```
+
+#### Cleaning up credentials
+
+We can use `vault lease revoke` to send an asynchronous revocation request before the TTL expires:
+
+```bash
+vault lease revoke database/creds/dev-read/TLCjNu5vT73r0QCW6X86f26Z
+All revocation operations queued successfully!
```

Comment:
Unfortunately this positive message will be output even if they cut-n-paste your lease id.